src/multiexp.rs

```diff
@@ -281,13 +281,15 @@ where
             }
         }
 
-        let (bss, skip) = bases.get();
-        let result = k.multiexp(pool, bss, Arc::new(exps), skip, n);
-
-        return Box::new(pool.compute(move || match result {
-            Ok(p) => Ok(p),
-            Err(e) => Err(SynthesisError::from(e)),
-        }));
+        let (bss, skip) = bases.clone().get();
+        match k.multiexp(pool, bss, Arc::new(exps.clone()), skip, n) {
+            Ok(p) => {
+                return Box::new(pool.compute(move || Ok(p)));
+            }
+            Err(e) => {
+                warn!("GPU Multiexp failed! Falling back to CPU... Error: {}", e);
```

Comment:
shouldn't this return an error?

Reply:
@dignifiedquire Depends on what you would expect it to do in case of a GPU failure. Fallback to CPU or return an error?

Reply:
ah right